### PR TITLE
fix module schema

### PR DESCRIPTION
**Pull request by bcoca**
no out of date hardcoded list anymore, uses 'current + 4' to
  set valid deprecation targets

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

test/schema

### CODE REVIEW

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py [hunk 1]
```diff
@@ -59,6 +60,12 @@ def collection_name(v, error_code=None):
     return v
 
 
+def deprecation_versions():
+    ''' Create a list of valid version for deprecation entries, current+4 '''
+    current = float('.'.join(__version__.split('.')[0:2]))
+    return Any(*[str(current + x / 100) for x in range(0, 5)])
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py]: This avoids using floats/division and uses `"""` for docstrings (follows PEP 8 style).

```suggestion
def deprecation_versions():
    """Create a list of valid version for deprecation entries, current+4"""
    major, minor = [int(version) for version in __version__.split('.')[0:2]]
    return Any(*['{0}.{1}'.format(major, minor + increment) for increment in range(0, 5)])
```

The use of floats/division can yield incorrect results:

```
'2.12' -> 2.12 -> ['2.12', '2.13', '2.14', '2.15', '2.16']
'2.13' -> ['2.13', '2.1399999999999997', '2.15', '2.1599999999999997', '2.17']
```

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py [hunk 2]
```diff
@@ -441,17 +448,10 @@ def deprecation_schema(for_collection):
     date_schema.update(main_fields)
 
     if for_collection:
-        version_schema = {
-            Required('removed_in'): version(for_collection),
-        }
+        version_schema = {Required('removed_in'): version(for_collection)}
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py]: I'd keep the original style here:

```suggestion
        version_schema = {
            Required('removed_in'): version(for_collection),
        }
```

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py [hunk 3]
```diff
@@ -441,17 +448,10 @@ def deprecation_schema(for_collection):
...
     else:
-        version_schema = {
-            # Only list branches that are deprecated or may have docs stubs in
-            # Deprecation cycle changed at 2.4 (though not retroactively)
-            # 2.3 -> removed_in: "2.5" + n for docs stub
-            # 2.4 -> removed_in: "2.8" + n for docs stub
-            Required('removed_in'): Any("2.12", "2.13", "2.14", "2.15", "2.16"),
-        }
+        version_schema = {Required('removed_in'): deprecation_versions()}
+
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py]: I'd keep the original style here:

```suggestion
        version_schema = {
            Required('removed_in'): deprecation_versions(),
        }
```

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]

File: test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py [hunk 4]
```diff
@@ -59,6 +60,12 @@ def collection_name(v, error_code=None):
     return v
 
 
+def deprecation_versions():
+    """Create a list of valid version for deprecation entries, current+4"""
+    major, minor = [int(version) for version in __version__.split('.')[0:2]]
+    return Any(*['{0}.{1}'.format(major, minor + increment) for increment in range(0, 5)])
```
Comment (webknjaz) [inline test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py:66]: How about 
```python
from decimal import Context, Decimal

def emit_version_range(ver, num):
    ctx = Context(3)  # x.xx
    for _ in range(num):
        ver = ver.next_plus(ctx)
        yield ver
```
and then
```suggestion
    ver = Decimal('.'.join(__version__.split('.')[:2]))
    return Any(*[str(v) for v in emit_version_range(ver, 5)])
```

(inspired by @nitzmahone's comments on Slack but it's probably overkill)

**Review by webknjaz — APPROVED** [review]

**Pull request merged**